automata/fa/dfa.py

```diff
@@ -1553,38 +1550,39 @@ def get_name_original(states: FrozenSet[DFAStateT]) -> DFAStateT:
             final_states=dfa_final_states,
         )
 
-    def show_diagram(self, path=None):
+    def iter_transitions(self):
+        return (
+            (from_, to_, symbol)
+            for from_, lookup in self.transitions.items()
+            for symbol, to_ in lookup.items()
+        )
+
+    def _get_input_path(
+        self, input_str
+    ) -> tuple[list[tuple[DFAStateT, DFAStateT, DFASymbolT], bool]]:
         """
-        Creates the graph associated with this DFA
+        Calculate the path taken by input.
+
+        Args:
+            input_str (str): The input string to run on the DFA.
+
+        Returns:
+            tuple[list[tuple[DFAStateT, DFAStateT, DFASymbolT], bool]]: A list
+            of all transitions taken in each step and a boolean indicating
+            whether the DFA accepted the input.
+
         """
-        # Nodes are set of states
+        state_history = [
```

Comment:
```suggestion
        state_history = list(self.read_input_stepwise(input_str, ignore_rejection=True))
```
You do a similar comprehension in a few places, I think this can be used to simplify things.